crates/db_views/src/post_report_view.rs

```diff
@@ -249,11 +246,25 @@ impl<'a> PostReportQueryBuilder<'a> {
 
     let (limit, offset) = limit_and_offset(self.page, self.limit);
 
-    let res = query
-      .order_by(post_report::published.asc())
+    query = query
+      .order_by(post_report::published.desc())
       .limit(limit)
-      .offset(offset)
-      .load::<PostReportViewTuple>(self.conn)?;
+      .offset(offset);
+
+    // If its not an admin, get only the ones you mod
+    let res = if !self.admin {
+      query
+        .inner_join(
+          community_moderator::table.on(
+            community_moderator::community_id
+              .eq(post::community_id)
+              .and(community_moderator::person_id.eq(self.my_person_id)),
+          ),
+        )
```

Comment:
You are repeating this code 4 times, maybe you can extract it into a function.

Reply:
Can't unfortunately, the function input for query is different for each of these, and it would be 2 pages of `BoxedSelectStatement<....tons of table joins>`

Reply:
Damn, i wish they could allow some kind of generics for diesel operations, would save us so much code.

Btw i just noticed that unit tests are failing.

Reply:
Ah, that pry has to do with my sort order change, I'll fix that in a sec.